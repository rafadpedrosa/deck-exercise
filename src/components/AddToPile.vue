<template>
  <div class="add-to-pile">
    <div>Add cards to the pile</div>
    <div>
      <input
        type="text"
        v-model="value"
        maxlength="2"
        @keyup.enter="emitValue"
      />
      <button class="primary-color" @click="emitValue">Add</button>
    </div>
  </div>
</template>

<script>
export default {
  data() {
    return {
      value: "",
    };
  },
  methods: {
    emitValue() {
      this.$emit("add", this.value);
    },
  },
};
</script>

<style scoped>
div {
  display: flex;
  justify-items: left;
}

.add-to-pile {
  display: flex;
  flex-direction: column;
}

input[type="text"] {
  border-radius: 4px 0 0 4px;
}

button {
  border-radius: 0 4px 4px 0;
}
</style>
