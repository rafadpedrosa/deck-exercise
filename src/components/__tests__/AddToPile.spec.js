import { shallowMount } from "@vue/test-utils";
import AddToPile from "../AddToPile";

describe("AddToPile.js", () => {
  it("Should create a snapshot of the AddToPile", () => {
    const wrapper = shallowMount(AddToPile, {});

    expect(wrapper.element).toMatchSnapshot();
  });

  it("Should emit sortByInput and sortByRotation after button click", () => {
    let expectedValue = "test";

    const wrapper = shallowMount(AddToPile, {
      propsData: {
        value: expectedValue,
      },
    });
    wrapper.vm.emitValue(); // call a method "naturally"

    expect(wrapper.emitted().add).toBeTruthy();
    expect(wrapper.emitted("add")[0]).toEqual([expectedValue]);
  });
});
