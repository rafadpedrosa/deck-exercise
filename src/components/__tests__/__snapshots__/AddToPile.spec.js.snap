// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddToPile.js Should create a snapshot of the AddToPile 1`] = `
<div
  class="add-to-pile"
>
  <div>
    Add cards to the pile
  </div>
   
  <div>
    <input
      maxlength="2"
      type="text"
    />
     
    <button
      class="primary-color"
    >
      Add
    </button>
  </div>
</div>
`;
